### DISCUSSION TITLE
Could not embed because more than one relationship was found -- Issues with view

### DISCUSSION THREAD

**Discussion by fj39** [supabase/supabase#27903] [Questions]
Hi! I've encountered a problem that I've been trying desperately to solve for a day now, and nothing seems to be working.
I have a fairly peculiar schema setup for my app. I'll try to explain in simple terms for context.

I've got tables for Championships, Drivers, Rounds, Sessions and Results.
Each `result` holds a reference to a `driver` and a `session`, as well as points scored in that session (along with other information).
Each `session` holds a reference to a `round`, which in turn is assigned to a `championship`.

I want to get a total of points scored by each driver in each round (from all of its sessions), however, I also need to account for drivers who took part in other rounds.

Now to the problem:
I've created a view with the following definition:
```plpgsql
create view
  public.round_driver_points as
with
  all_rounds_drivers as (
    select distinct
      r.id as round_id,
      d_1.id as driver_id
    from
      rounds r
      cross join drivers d_1
  ),
  filtered_sessions as (
    select
      s.id as session,
      s.round,
      r.driver,
      r.points
    from
      sessions s
      join results r on s.id = r.session
  ),
  driver_round_points as (
    select
      fs.round,
      fs.driver,
      coalesce(sum(fs.points), 0::numeric) as total_points
    from
      filtered_sessions fs
    group by
      fs.round,
      fs.driver
  ),
  final_points as (
    select
      ard.round_id,
      ard.driver_id,
      coalesce(drp.total_points, 0::numeric) as points,
      case
        when drp.total_points is null then true
        else false
      end as dns
    from
      all_rounds_drivers ard
      left join driver_round_points drp on ard.round_id = drp.round
      and ard.driver_id = drp.driver
  )
select
  fp.round_id,
  fp.driver_id as round_driver_id,
  fp.points,
  fp.dns
from
  final_points fp
  join drivers d on fp.driver_id = d.id
order by
  fp.round_id,
  d.id;
```

Which actually works great in the Supabase SQL Query - it does return the values I expected to see.
However, when I want to get this data in my app using
```js
const { data: overview, error } = await sb.from('round_driver_points').select('*, round_driver_id(*), round_id(*)')
```

I'm getting an error:
```
code: "PGRST201"
details: (6) [{…}, {…}, {…}, {…}, {…}, {…}]
hint: "Try changing 'round_driver_id' to one of the following: 'collisions!collisions_driver_fkey', 'collisions!collisions_other_driver_fkey', 'dotd_options!dotd_options_driver_fkey', 'driver_teams!driver_teams_driver_fkey', 'laps!laps_driver_fkey', 'results!results_driver_fkey'. Find the desired relationship in the 'details' key."
message: "Could not embed because more than one relationship was found for 'round_driver_points' and 'round_driver_id'"
```

I've tried so many different ways to resolve it. I've no idea what I'm doing wrong anymore, and I've read every issue I could find remotely related to this, but it seems that my use case is a bit more complex than those others.

Any help would be massively appreciated - thank you kindly.

**Comment by GaryAustin1** [comment]
The warning is here are multiple paths thru your foreign key relations to the collision table I believe.
Note that with Views all the relations are done using the base table foreign keys and not relationships you might use or have in the view.
![image](https://github.com/supabase/supabase/assets/54564956/11f249f6-a8e2-4375-bc65-d1341114a166)

Not sure how much help that is.  Normally the suggestions on how to fix work by using one of them.

**Reply by fj39** [reply]
Thanks for the reply!

I'm not very experienced in working with Postgres. Am I understanding correctly then that the issue is the amount of tables I'm trying to work through in my WITH statements?

**Reply by GaryAustin1** [reply]
No.  You say the view query works as you expect.  The problem is in your JS query PostgREST (the REST API) you are trying to join based on info in the View.  But PostgREST only joins based on foreign key references in your actual tables.
https://postgrest.org/en/v12/references/api/resource_embedding.html#foreign-key-joins-on-views
Covers most of it.  I'm not much help with a solution for you, just pointing out a limitation of using views and then also joining on them in JS.
